### PR TITLE
[Pornhub] Fix private video detection

### PR DESCRIPTION
**Pull request by kidol**
Forgot to test private video in #9964, so it got broke...

Example url: http://www.pornhub.com/view_video.php?viewkey=ph56fd731fce6b7

### CODE REVIEW

**Comment by dstftw** [comment]
Add a test.

**Comment by kidol** [comment]
`python test/test_download.py TestDownload.test_PornHub`
only runs the first test... How to run all?

>Ran 1 test in 5.155s

**Comment by dstftw** [comment]
`TestDownload.test_PornHub_<number>`.

File: youtube_dl/extractor/pornhub.py [hunk 1]
```diff
@@ -107,7 +107,7 @@ def _real_extract(self, url):
         webpage = self._download_webpage(req, video_id)
 
         error_msg = self._html_search_regex(
-            r'(?s)<div[^>]+class=(["\']).*?\bremoved\b.*?\1[^>]*>(?P<error>.+?)</div>',
+            r'(?s)<div[^>]+class=(["\']).*?\b(removed|userMessageSection)\b.*?\1[^>]*>(?P<error>.+?)</div>',
```
Comment (dstftw) [inline youtube_dl/extractor/pornhub.py]: Do not capture group if you don't use it.

**Comment by kidol** [comment]
Thanks but I still don't know how the test works. Test 1 & 2 work with `TestDownload.test_PornHub_<number>`, how do the rest work?

i.e.:

```python
{
    # removed at the request of the copyright owner
    'url': 'http://www.pornhub.com/view_video.php?viewkey=788152859',
    'only_matching': True,
},
```

What does only_matching mean?

**Comment by dstftw** [comment]
`only_matched` means exactly that: only test URL to match `_VALID_URL`. It's tested in `test_all_urls.py`.

**Comment by kidol** [comment]
>only_matched means exactly that: ...

Ok the comments confused me, they suggest something else.

Anyway, when I look at other tests (YouTube), I don't see any tests which check for specific error messages like "this video is private" etc.

What do you want me to do?

**Comment by dstftw** [comment]
Add only matching test.

**Comment by kidol** [comment]
I don't get it.

This url structure

>view_video.php?viewkey=phxxxxxxxxxxx

is already covered in a matching test.

?

**Comment by dstftw** [comment]
So what?

**Comment by kidol** [comment]
You want me to add this:

```python
{
        # video is private
        'url': 'http://www.pornhub.com/view_video.php?viewkey=ph56fd731fce6b7',
        'only_matching': True,
}
```

But why? It's same as existing test:

```python
{
        # removed by uploader
        'url': 'http://www.pornhub.com/view_video.php?viewkey=ph572716d15a111',
        'only_matching': True,
}
```

**Comment by dstftw** [comment]
Because
>Forgot to test private video in #9964, so it got broke...

**Comment by kidol** [comment]
Yes this was related to the regex. Has nothing to do with url structure.

**Comment by kidol** [comment]
Now there are 2 matching tests which are equivalent and for what?
I just want to understand... :weary:

**Comment by dstftw** [comment]
All these tests are for reference to keep track of possible "unavailable" URLs to do manual testing after some changes applied in extractor's code. Doing real testing for them during travis build is just a time waste since it does not really matter whether unavailability will be detected or not. What really matters is to test "available" URLs are really available.

**Comment by kidol** [comment]
Ok thanks.

**Pull request closed without merging**